src/psyclone/transformations.py

```diff
@@ -353,6 +353,20 @@ def __init__(self, omp_schedule="static"):
         self.omp_schedule = omp_schedule
         Transformation.__init__(self)
 
+    def _validate(self, node):
+        ''' Perform validation checks before applying the transformation'''
+        # Check that the supplied node is a Loop
+        from psyclone.psyGen import Loop
+        if not isinstance(node, Loop):
+            raise TransformationError("Cannot apply an OpenMP Loop "
+                                      "directive to something that is "
+                                      "not a loop")
+        # Check we are not a sequential loop
+        if node.loop_type == 'colours':
+            raise TransformationError("Error in "+self.name+" transformation. "
+                                      "The target loop is over colours and "
+                                      "must be computed serially.")
+
     def apply(self, node, reprod=None):
         '''Apply the OMPLoopTrans transformation to the specified node in a
```

Comment:
Please update the docstring with the interface to this routine.

Reply:
Done.